src/ccl_power.c

```diff
@@ -407,277 +431,130 @@ static void ccl_cosmology_compute_power_class(ccl_cosmology * cosmo, int * statu
   // to avoid writing and reading .ini files for every call
   int parser_length = 20;
   int init_arr[7]={0,0,0,0,0,0,0};
+  int init_parser=0;
   if (parser_init(&fc,parser_length,"none",errmsg) == _FAILURE_) {
     *status = CCL_ERROR_CLASS;
     ccl_cosmology_set_status_message(cosmo, "ccl_power.c: ccl_cosmology_compute_power_class(): parser init error:%s\n", errmsg);
-    return;
   }
 
-  ccl_fill_class_parameters(cosmo,&fc,parser_length, status);
+  if(*status==0)
+    ccl_fill_class_parameters(cosmo,&fc,parser_length, status);
 
-  if (*status != CCL_ERROR_CLASS)
+  if (*status==0) {
+    init_parser=1;
```

Comment:
The way this is setup if we fail to fill the class parameters, the parser will not be freed.

Reply:
Argh, sorry. Good catch. This was meant to go in the previous block

Reply:
still needs to be changed. I see a refresh at the top. I will check back in a bit.

Reply:
Mmh, I've changed this in the last commit